app/carbonapi/app.go

```diff
@@ -391,15 +391,20 @@ func (app *App) deferredAccessLogging(accessLogger *zap.Logger, r *http.Request,
 	if err != nil {
 		accessLogger.Error("could not marshal access log details", zap.Error(err))
 	}
-
+	var logMsg string
```

Comment:
This is a change in logic. Now we can call the function to log an error but it will be logged as "request served".
Also, not all requests that all not 2xx are failures. 
Is this what we want?

Reply:
> Could you please explain the reasoning behind this?
Yes, of course. As I already mentioned in the PR description, we are getting lots of error logs caused by client errors. Error level should simply be for the cases we experience an issue inside the system.
As an example, if we cannot parse results from go-carbon, it is an error. If we get timed out while gathering the results, it is an error. But if the user provides a wrong value for a field, it's not.

> Now we can call the function to log an error but it will be logged as "request served"
Yes you can. But we never do in the code even though it seems right when we have written the response with 2xx to log "request served". It just doesn't make sense to call the function like that.
Anyways, the whole way of logging with this `logAsError` is not desirable in my opinion. But I didn't want to have a bigger change in the logging logic in carbonapi. This can be done in another PR or some following commits (I prefer the former). 

> Also, not all requests that all not 2xx are failures.
From what I observed, all non-2xx cases are failures in the code (we only have 2xx, 4xx, and 5xx, don't we?).

Reply:
> Also, not all requests that all not 2xx are failures.
This is fundamentally true. Logging otherwise is treating HTTP codes wrongfully.

IMO, we can remove `logMsg` altogether. Opinions?

Reply:
I'm not sure what you're proposing. What will we log then?